eplmp-server/eplmp-server-rest/src/main/java/org/polarsys/eplmp/server/rest/util/SearchQueryParser.java

```diff
@@ -239,19 +250,18 @@ public static PartSearchQuery parsePartStringQuery(String workspaceId, Multivalu
 
     }
 
-    private static List<SearchQuery.AbstractAttributeQuery> parseAttributeStringQuery(String attributeQuery) {
-        List<SearchQuery.AbstractAttributeQuery> pAttributes = new ArrayList<>();
-        String[] attributesString = attributeQuery.split(ATTRIBUTES_DELIMITER);
-
-        for (String attributeString : attributesString) {
+    private static List<SearchQuery.AbstractAttributeQuery> parseAttributeStringQuery(JsonArray attributeQuery) {
 
-            int firstColon = attributeString.indexOf(ATTRIBUTES_SPLITTER);
-            String attributeType = attributeString.substring(0, firstColon);
-            attributeString = attributeString.substring(firstColon + 1);
+        List<SearchQuery.AbstractAttributeQuery> pAttributes = new ArrayList<>();
+        for (JsonValue attributeString : attributeQuery) {
```

Comment:
Even if it does not change fundamentally why not use  attributeQuery.getValuesAs(JsonObject.class) just to prevent the warning message due to the cast operation?